src/community/ogcapi/ogcapi-coverages/src/main/java/org/geoserver/ogcapi/v1/coverages/CollectionsDocument.java

```diff
@@ -99,6 +96,35 @@ public CollectionDocument next() {
         };
     }
 
+    private CollectionDocument getCollectionDocument(
+            CoverageInfo coverage, CloseableIterator<CoverageInfo> coverages) {
+        try {
+            List<String> crs =
+                    CoveragesService.getCoverageCRS(coverage, Collections.singletonList("#/crs"));
+            CollectionDocument collection = new CollectionDocument(geoServer, coverage, crs);
+            return collection;
+        } catch (Exception e) {
+            if (skipInvalid) {
+                LOGGER.log(Level.WARNING, "Skipping coverage type " + coverage.prefixedName());
+                if (coverages.hasNext()) {
```

Comment:
This works if there is one collection failing and the next is working. But it should work even if 9 out of 10 of collections are broken. The code should look as follows (restructuring this method and the one calling it):

```java
while(coverages.hasNext()) {
   try {
        CoverageDocument next = getCollectionDocument();
        ...
   } catch(Exception e) {
         if (skipExceptions) // log and allow next iteration in while loop
         else rethrow ...
   }
}

Reply:
Ah, recursion actually fetches the next one. Works, but it's limited by the max stack depth

Reply:
Changed to non-recursion version